src/server_manager/web_app/app.ts

```diff
@@ -386,6 +477,19 @@ export class App {
   }
 
   private connectToDigitalOcean() {
+    const accessToken = this.digitalOceanTokenManager.getStoredToken();
+    if (accessToken) {
```

Comment:
Seems like this whole block is basically duplicated in `start` - and then later in this same function!

If it helps, I noticed this while looking at all the calls to `syncServersToDisplay` - seems like there should only be 2-3 places where that's called, right?

Reply:
I tried to refactor it into a separate method, but if you notice closely, the logic is slightly different on every call. Please let me know if you have ideas on how to improve this.

Reply:
This seems like something to tackle in the future.